new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItTraefikLoadBalancer.java

```diff
@@ -0,0 +1,353 @@
+import static oracle.weblogic.kubernetes.actions.TestActions.getServiceNodePort;
+import static oracle.weblogic.kubernetes.actions.TestActions.uninstallTraefik;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkPodReady;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.checkServiceExists;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createDockerRegistrySecret;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createDomainAndVerify;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createSecretWithUsernamePassword;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.createTraefikIngressForDomainAndVerify;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.CommonTestUtils.installAndVerifyTraefik;
+import static oracle.weblogic.kubernetes.utils.TestUtils.getNextFreePort;
+import static oracle.weblogic.kubernetes.utils.TestUtils.verifyClusterMemberCommunication;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests related to WebLogic domain traffic routed by traefik loadbalancer.
+ */
+@DisplayName("Test traefik loadbalancing with multiple WebLogic domains")
+@IntegrationTest
+public class ItTraefikLoadBalancer {
+
+  private static String opNamespace = null;
+  private static String domainNamespace = null;
+  private static String traefikNamespace = null;
+
+  private static int nodeportshttp;
+
+  private static HelmParams traefikHelmParams = null;
+
+  private static final String IMAGE = MII_BASIC_IMAGE_NAME + ":" + MII_BASIC_IMAGE_TAG;
+
+  private final String wlSecretName = "weblogic-credentials";
+
+  private static Path clusterViewAppPath;
+  private static LoggingFacade logger = null;
+
+  /**
+   * Assigns unique namespaces for operator and domains. Installs operator.
+   *
+   * @param namespaces injected by JUnit
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(3) List<String> namespaces) {
+    logger = getLogger();
+
+    logger.info("Assign a unique namespace for operator");
+    opNamespace = namespaces.get(0);
+
+    logger.info("Assign a unique namespace for WebLogic domains");
+    domainNamespace = namespaces.get(1);
+
+    logger.info("Assign a unique namespace for traefik");
+    traefikNamespace = namespaces.get(2);
+
+    // install operator and verify its running in ready state
+    logger.info("Installing operator");
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // get a free web and websecure ports for traefik
+    nodeportshttp = getNextFreePort(30380, 30405);
```

Comment:
Can you let Kubernetes pick up the nodeport for Traefik then get the assigned nodeport through getServiceNodePort() method? In parallel run, the predefined nodeport may got port conflicts.

Reply:
fixed.

Reply:
Good point, Thanks